### PR TITLE
Marshall Houston 1708-be

### PR DESCRIPTION
**Pull request by marshallhouston**
rails_challenge_2

### CODE REVIEW

**Comment by marshallhouston** [comment]
@icorson3 @s-espinosa

I realize now that I didn't use the user model at all; I added a username column on the review instead of adding a foreign_key that referenced the users.

To tie users with reviews, here are some of the changes I would make.
- create migration to add a column to the reviews table. it would be a reference to users so that it has a user_id column
- in the User model, I would add ```has_many :reviews```
- in the Review model, I would add ```belongs_to :user```

This would then allow me to call .user on a review to get the user object; I could then call .name to find the name of the user that was associated with that review.

File: app/models/review.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Review < ApplicationRecord
+  belongs_to :book
```
Comment (icorson3) [inline app/models/review.rb:2]: `belongs_to :user`

File: spec/features/books/user_can_see_a_book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe "When a user visits a book show page" do
+  scenario "I see the book title and reivews" do
+    book = Book.create(title: "Book1")
+    book.reviews.create(username: "grayton", body: "i really liked this book", rating: 5)
+    book.reviews.create(username: "grayton2", body: "ibook", rating: 3)
+    book.reviews.create(username: "grayton3", body: "i", rating: 1)
```
Comment (icorson3) [inline spec/features/books/user_can_see_a_book_spec.rb:8]: would want to attach a user to a review (saw that you got that later) `book.reviews.create( body: "hi", rating: 5, user: user)`

File: spec/models/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    before :each do
+      @book = Book.create(title: "Book1")
+      @book.reviews.create(username: "grayton", body: "i really liked this book", rating: 5)
+      @book.reviews.create(username: "grayton2", body: "ibook", rating: 3)
+      @book.reviews.create(username: "grayton3", body: "i", rating: 1)
+    end
+
+    describe "#average_rating" do
+      it "calculates the average rating for all reviews for a book" do
```
Comment (icorson3) [inline spec/models/book_spec.rb:21]: 🎉

**Comment by icorson3** [comment]
added a few comments. see that you saw your mistake...testing is solid 🎉

**Pull request closed without merging**